src/Agent.Worker/Container/ContainerInfo.cs

```diff
@@ -68,6 +98,23 @@ public List<MountVolume> MountVolumes
             }
         }
 
+        public List<PortMapping> PortMappings
+        {
+            get
+            {
+                if (_portMappings == null)
+                {
+                    _portMappings = new List<PortMapping>();
+                }
+
+                return _portMappings;
+            }
+            set
+            {
+                _portMappings = value;
```

Comment:
we don't normally expose `Set`

Reply:
This was exposed because it may need to be reassigned at runtime. If the ports are not given in the build definition, they are assigned a random port by the host. I shell out to `docker port` to find the dynamic port assignment after the container is started, parse, and set the new mapping back so that it can used to provide access (exposed over agent.services.nginx.ports.80, for eg). 

I will investigate if there is a better way to handle ports here

Reply:
For that case, normally we don't expose the store object. we add method to allow consumer to edit the object, ex: `void ContainerInfo.AddPortMapping()`